runtime/compiler/net/ServerStream.hpp

```diff
@@ -139,19 +139,25 @@ class ServerStream : CommunicationStream
    std::tuple<T...> readCompileRequest()
       {
       readBlocking(_cMsg);
-      if (_cMsg.type() == JITServer::MessageType::clientTerminate)
+      if (_cMsg.type() == MessageType::clientTerminate)
          {
          uint64_t clientId = std::get<0>(getRecvData<uint64_t>());
-         throw JITServer::StreamCancel(_cMsg.type(), clientId);
+         throw StreamCancel(_cMsg.type(), clientId);
          }
-      if (_cMsg.version() != 0 && _cMsg.version() != getJITaaSVersion())
+      else if (_cMsg.type() == MessageType::compilationAbort)
          {
-         throw StreamVersionIncompatible(getJITaaSVersion(), _cMsg.version());
+         throw StreamCancel(_cMsg.type());
+         }
+      else if (_cMsg.type() != MessageType::compilationRequest)
```

Comment:
Why did you feel that you needed to reorder the messages?

Reply:
I felt like the first three are msg type checks. Reordered so that I can use the `else if` instead of three separate `if`s.